### DISCUSSION TITLE
Add the 'Safely Shut Down' command for SMuRF to the protocol.

### DISCUSSION THREAD

**Discussion by yuhanwyhan** [simonsobs/sorunlib#121] [Ideas]
Hi, is it possible to add in one command to safely shut down the SMuRF RF/DC power (S.all_off) and amps power (clear the biasing voltage)?

**Comment by BrianJKoopman** [comment]
That sounds possible. What's the corresponding command in the [pysmurf-controller agent](https://socs.readthedocs.io/en/main/agents/pysmurf-controller.html)? If one doesn't exist, it'll need to be added there first.

**Reply by yuhanwyhan** [reply]
I think it is called all_off https://socs.readthedocs.io/en/main/_modules/socs/agents/pysmurf_controller/agent.html#PysmurfController.all_off

**Reply by BrianJKoopman** [reply]
Thanks, issue created here: https://github.com/simonsobs/sorunlib/issues/122
